### PR TITLE
[CHK-1433] add npg wallet api

### PR DESCRIPTION
**Pull request by ciuffagianluca**
Add new group of wallet api to handle npg interaction

### List of changes

Add notifiy api exposed to npg, to update wallet status when a payment is reqeusted

### Motivation and context

Wallet  workflow implementation requires to expose a notify api to the npg

### Type of changes

- [x] Add new resources
- [ ] Update configuration to existing resources
- [ ] Remove existing resources

### Does this introduce a change to production resources with possible user impact?

- [ ] Yes, users may be impacted applying this change
- [ ] No

### Does this introduce an unwanted change on infrastructure? Check terraform plan execution result

- [ ] Yes
- [ ] No

### Other information

---

### CODE REVIEW

File: src/domains/wallet-app/api/npg/v1/_openapi.json.tpl [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+              "schema": {
+                "type": "string",
+                "format": "uuid"
+              },
+              "required": true,
+              "description": "Unique identifier of the request"
+            }
+          ],
+          "requestBody": {
+            "description": "Notifiy wallet",
```
Comment (giovanniberti) [inline src/domains/wallet-app/api/npg/v1/_openapi.json.tpl:48]:
```suggestion
            "description": "Notify wallet",
```

**Review by giovanniberti — COMMENTED** [review]
typo

File: src/domains/wallet-app/api/npg/v1/_openapi.json.tpl [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+            "code": {
+              "type": "string"
+            },
+            "description": {
+              "type": "string"
+            }
+          }
+        }
+      }
+    }
```
Comment (pietro-tota) [inline src/domains/wallet-app/api/npg/v1/_openapi.json.tpl:137]: Should we use `ProblemJson` object to return problems to the caller as done in ecommerce or also [here](https://github.com/pagopa/pagopa-infra/blob/e1037bda2baeeec28f2c033e06d2e843cce240a1/src/domains/wallet-app/api/wallet-service/v1/_openapi.json.tpl#L334-L374) in the wallet?

File: src/domains/wallet-app/api/npg/v1/_base_policy.xml.tpl [hunk 3]
```diff
@@ -0,0 +1,15 @@
+<policies>
+    <inbound>
+      <base />
+      <set-backend-service base-url="https://${hostname}/pagopa-wallet-notification-service" />
```
Comment (pietro-tota) [inline src/domains/wallet-app/api/npg/v1/_base_policy.xml.tpl:4]:
```suggestion
      <set-backend-service base-url="https://${hostname}/pagopa-wallet-service" />
```

Should it be the same backend service url as the one exposed for the wallet?

**Review by pietro-tota — COMMENTED** [review]
Some comment

File: src/domains/wallet-app/api/npg/v1/_openapi.json.tpl [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+                "schema": {
+                  "$ref": "#/components/schemas/NotificationRequest"
+                }
+              }
+            },
+            "required": true
+          },
+          "responses": {
+            "204": {
+              "description": "Payment reqeust sent successfully"
```
Comment (pietro-tota) [inline src/domains/wallet-app/api/npg/v1/_openapi.json.tpl:60]: ### 
```suggestion
              "description": "Notification handled successfully"
```

**Review by pietro-tota — COMMENTED** [review]
Some comment

File: src/domains/wallet-app/api/npg/v1/_openapi.json.tpl [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+              "content": {
+                "application/json": {
+                  "schema": {
+                    "$ref": "#/components/schemas/ErrorResponse"
+                  }
+                }
+              },
+            },
+            "405": {
+              "description": "Invalid input"
```
Comment (Fernando-Granato) [inline src/domains/wallet-app/api/npg/v1/_openapi.json.tpl:83]: This way you only return a string, wouldn’t it be better to return an ErrorResponse object ?

File: src/domains/wallet-app/api/npg/v1/_base_policy.xml.tpl (hunk 3 above)
Comment (ciuffagianluca) [reply src/domains/wallet-app/api/npg/v1/_base_policy.xml.tpl:4]: I push this different url to have clarification about something not clear for me.
This api is declared in a new api group in wallet project, so my doubt is if we have to maintain same backend service url or it should be different.

Comment (ciuffagianluca) [reply src/domains/wallet-app/api/npg/v1/_base_policy.xml.tpl:4]: Solved my doubt, I'll push your suggestion

File: src/domains/wallet-app/api/npg/v1/_openapi.json.tpl (hunk 5 above)
Comment (ciuffagianluca) [reply src/domains/wallet-app/api/npg/v1/_openapi.json.tpl]: Removed this code and used problem json to describe the error

File: src/domains/wallet-app/api/npg/v1/_openapi.json.tpl [hunk 6]
```diff
@@ -0,0 +1,162 @@
...
+                }
+              }
+            },
+            "required": true
+          },
+          "responses": {
+            "204": {
+              "description": "Notification handled successfully"
+            },
+            "400": {
```
Comment (Fernando-Granato) [inline src/domains/wallet-app/api/npg/v1/_openapi.json.tpl:62]: You can add different body examples for error cases ?
```suggestion
            "400": {
            example:
                value:
                  type: https://example.com/problem/
                  title: string
                  status: 404
                  detail: Element not fount
```

**Review by Fernando-Granato — APPROVED** [review]

File: src/domains/wallet-app/04_apim_wallet.tf [hunk 7]
```diff
@@ -67,4 +67,31 @@ module "apim_wallet_service_api_v1" {
   xml_content = templatefile("./api/wallet-service/v1/_base_policy.xml.tpl", {
     hostname = local.wallet_hostname
   })
+}
+
+module "apim_wallet_service_notification_api_v1" {
+  source = "git::https://github.com/pagopa/terraform-azurerm-v3.git//api_management_api?ref=v6.3.0"
+
+  name                  = format("%s-service-api", local.project)
```
Comment (infantesimone) [inline src/domains/wallet-app/04_apim_wallet.tf:75]: specify `notifications` in the api group name to distinguish it from the other

**Review by infantesimone — CHANGES_REQUESTED** [review]

**Review by infantesimone — APPROVED** [review]

**Pull request merged**